### PR TITLE
Add assignments and constraints for `AccountDoesNotExist`

### PR DESCRIPTION
**Pull request by silathdiir**
### Summary

1. Main constraints for type 1 non-existence proof:
```
direction == 0
key != other_key
(poseidon) lookup for [1, other_key, other_key_hash]
```

2. Main constraints for type 2 non-existence proof:
```
old_hash == 0
new_hash == 0
```

3. Conditions for type 1 non-existence proof:
```
next_path == Start
current_segment == AccountLeaf0
```

4. Conditions for type 2 non-existence proof:
```
next_path == Start
current_segment == AccountTrie
```

5. Update `assign` and constrain in both `configure_common_path` and `configure_empty_account` functions.

### Test

Test case `nonexisting_type_1` and `nonexisting_type_2` could pass (and remaining 4 failed cases, same as target branch).

### CODE REVIEW

**Comment by huwenqing0606** [comment]
Just curious: in configure common path why we delete the original hash constraints like hash=poisedon(left, right)?

File: src/gadgets/mpt_update.rs [hunk 1]
```diff
@@ -1367,7 +1376,112 @@ fn configure_keccak_code_hash<F: FieldExt>(
...
+    for variant in SegmentType::iter() {
+        let conditional_constraints = |cb: &mut ConstraintBuilder<F>| match variant {
+            SegmentType::Start => {}
+            SegmentType::AccountTrie => {
+                // Check if next path is start for type 2 non-existence.
+                cb.condition(config.path_type.next_matches(&[PathType::Start]), |cb| {
+                    configure_non_existing_type2(cb, config)
+                });
+            }
+            SegmentType::AccountLeaf0 => {
```
Comment (huwenqing0606) [inline src/gadgets/mpt_update.rs:1620]: Do we have to put a condition saying next `PathType` is `Start` here?

Comment (silathdiir) [reply src/gadgets/mpt_update.rs:1620]: Fix to constrain current path type is `Common` and next is `Start` for type 1 non-existence proof in commit [8143bc1](https://github.com/scroll-tech/mpt-circuit/pull/40/commits/8143bc1f1e1943c1491f91e468140ce5b2e45e3d).

**Review by huwenqing0606 — COMMENTED** [review]
LGTM. Since type 1 and type 2 non-existence proof can refer to both empty-account case or insert/delete case, I will update the spec to make a clearer description of this convention

**Review by z2trillion — COMMENTED** [review]
looks great. thanks!

**Comment by huwenqing0606** [comment]
I just realized one thing: if we constrain type 1 and 2 non-existence inside `PathType::Common`, and then we do the same constrains inside `MPTProofType::AccountDoesNotExist`, are we duplicating these constraints? Maybe I missed something...

**Pull request merged**